src/main/java/seedu/staffsnap/logic/commands/DeleteInterviewCommand.java

```diff
@@ -0,0 +1,95 @@
+package seedu.staffsnap.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.staffsnap.logic.parser.CliSyntax.PREFIX_INTERVIEW;
+import static seedu.staffsnap.model.Model.PREDICATE_HIDE_ALL_APPLICANTS;
+import static seedu.staffsnap.model.Model.PREDICATE_SHOW_ALL_APPLICANTS;
+
+import java.util.List;
+
+import seedu.staffsnap.commons.core.index.Index;
+import seedu.staffsnap.commons.util.ToStringBuilder;
+import seedu.staffsnap.logic.Messages;
+import seedu.staffsnap.logic.commands.exceptions.CommandException;
+import seedu.staffsnap.model.Model;
+import seedu.staffsnap.model.applicant.Applicant;
+import seedu.staffsnap.model.interview.Interview;
+
+/**
+ * Deletes an applicant's specified interview identified using it's displayed index from the applicant book.
+ */
+public class DeleteInterviewCommand extends Command {
+
+    public static final String COMMAND_WORD = "deletei";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the applicant's specified interview identified by the index number "
+            + "used in the displayed applicant list.\n"
+            + "Parameters: INDEX (must be a positive integer) \n"
+            + PREFIX_INTERVIEW + "INTERVIEW_INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1 i/2";
+
+    public static final String MESSAGE_DELETE_INTERVIEW_SUCCESS = "Deleted interview from Applicant: %1$s";
+
+    private final Index targetApplicantIndex;
+    private final Index targetInterviewIndex;
+
+    /**
+     * Creates a DeleteInterviewCommand to delete the interview at the
+     * specified {@code targetInterviewIndex} from the applicant at the
+     * {@code targetApplicantIndex} in the displayed Applicant list.
+     */
+    public DeleteInterviewCommand(Index targetApplicantIndex, Index targetInterviewIndex) {
+        this.targetApplicantIndex = targetApplicantIndex;
+        this.targetInterviewIndex = targetInterviewIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Applicant> lastShownList = model.getFilteredApplicantList();
+
+        if (targetApplicantIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
Is it meant to give a different error when number is negative vs number is too large, I think error should be the same if not have to handle 2 different cases message

Reply:
The two separate errors are intended, as one pertains to incorrect command format, the other pertains to an invalid argument range. 

This encourages **greater cohesion** and **reduced coupling** according to the Separation of Concerns principle found in https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/principles.html#separation-of-concerns-principle.

The command parser should only parse out the input string according to the command format, whereas the the invalid argument range should only be caught when the command is being executed.

In any case, I am happy to handle two different cases if it entails greater code modularity 😃

Reply:
Ok, found it a bit weird if user keys in negative number the error is invalid command format but if they key in too large number, its invalid index since negative number should also be an invalid index error but this way should be fine also.